src/Modules/AuthModule/components/Register/Register.tsx

```diff
@@ -1,5 +1,403 @@
-import React from "react";
+import React, { useContext, useState } from "react";
+import { ErrorMessage } from "@hookform/error-message";
+import axios from "axios";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+
+import Images from "../../../ImageModule/components/Images/Images";
+
+import BaceUrlContext from "../../../../Context/BaceUrlContext";
 
 export default function Register() {
-  return <>Register</>;
+  // instance from use navigate
+  const navigate = useNavigate();
+  // image input state
+  const [image, setImage] = useState(null);
+  // hide pass or show state
+  let [hidePassInInpt, setHidePassInInpt] = useState(true);
```

Comment:
The name here is misleadding, as it indicates to the reader that that `hidePassInpt` is a function not a state variable.
We can change it to something like `[isPasswordHidden,setIsPasswordHidden]`